### PR TITLE
Add benchmarking for Tree Building exercise

### PR DESCRIPTION
**Pull request by TeaDrivenDev**
This is what user-runnable benchmarking for the Tree Building exercise could look like, as discussed in #736. The original code (in `TreeBuildingBaseline.fs`) and the student's work (in `TreeBuilding.fs`) are benchmarked using the input data for all the unit tests.

A few things here:
- To run the benchmarks for the two solutions with the same logic, the `Record` type had to be extracted into its own module `TreeBuildingTypes` that is accessible from all other modules. This does not add any new restrictions, as the type already had to remain unchanged before, because it was used in the unit tests.
- The test file should probably get a new version; I'm not sure what it should be, though. 1.1.0?
- The instructions should probably be expanded, elaborating more on the performance "requirement" and potentially the benchmark.
- I think BenchmarkDotNet allows changing the columns displayed in the table. It might make sense to hide most of the default columns, as they're not helpful for our use case.

### CODE REVIEW

**Comment by ErikSchierboom** [comment]
I really like this! It's great to have at least one exercise that focuses on performance, and the setup you've made works well. 

> To run the benchmarks for the two solutions with the same logic, the Record type had to be extracted into its own module TreeBuildingTypes that is accessible from all other modules.

This is fine, and in fact teaches the student about sharing types between files :)

> The test file should probably get a new version; I'm not sure what it should be, though. 1.1.0?

I'm thinking this being a very different version of the exercise we should be bumping the version to 2.0.0 perhaps?

> The instructions should probably be expanded, elaborating more on the performance "requirement" and potentially the benchmark.

Updating the README is a great idea. I'm thinking perhaps adding some comments in the code too, for the students that don't read the README (that's me ;))?

> I think BenchmarkDotNet allows changing the columns displayed in the table. It might make sense to hide most of the default columns, as they're not helpful for our use case.

Feel free to look into this!

Overall I love this PR! Thanks a ton for doing this.

Note to self: before merging this, we'll need to remove this exercise from the core exercises.

**Comment by TeaDrivenDev** [comment]
- Reduced summary columns to `Mean` and `Allocated`
- Added/updated file version information
- Expanded instructions
- Slight performance "deprovement"

File: exercises/tree-building/README.md [hunk 1]
```diff
@@ -1,5 +1,21 @@
 # Tree Building
 
+This exercise is different from most others in that you already start with
```
Comment (ErikSchierboom) [inline exercises/tree-building/README.md]: Great introduction!

File: exercises/tree-building/TreeBuilding.fs [hunk 2]
```diff
@@ -1,26 +1,35 @@
-﻿module TreeBuilding
+﻿// This is the file you need to modify for your own solution.
+// The unit tests will use this code, and it will be used by the benchmark tests
+// for the "Mine" row of the summary table.
 
-type Record = { RecordId: int; ParentId: int }
-type Tree = 
+// Remember to not only run the unit tests for this exercise, but also the
+// benchmark tests using `dotnet run -c Release`.
+// Please refer to the instructions for more information about the benchmark tests.
```
Comment (ErikSchierboom) [inline exercises/tree-building/TreeBuilding.fs]: Excellent addition.

**Comment by ErikSchierboom** [comment]
With #742 merged, this has now also been merged. Thanks a ton @TeaDrivenDev! Really excited about this change. 🎉

**Comment by robkeim** [comment]
Cool stuff @TeaDrivenDev!

**Pull request merged**